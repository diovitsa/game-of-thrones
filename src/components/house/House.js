import React, { useState } from 'react';
import { Text, View } from 'react-native';
import CharacterList from '../character-list/CharacterList';

function House({ house }) {
  const [isOpen, setOpen] = useState(false);

  const houseDetails = isOpen
    ? <View>
      <Text>Words: {house.words || '-'}</Text>
      <CharacterList house={house}/>
    </View>
    : null;

  return (
    <View>
      <Text style={{ color: 'blue', fontSize: 20 , textAlign: 'center'}} onPress={() => setOpen(!isOpen)}>{house.name}</Text>
      {houseDetails}
    </View>
  );
}

export default House;