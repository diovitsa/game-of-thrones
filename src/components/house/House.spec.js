import React from 'react';
import House from './House';
import TestRenderer, { act } from 'react-test-renderer';
import { Text, View } from 'react-native';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'


describe('House', () => {
  let component;
  let wrapper;

  const defaultProps = {
    house: {
      words: 'testWords',
      name: 'test house name'
    }
  };

  const middlewares = [];
  const mockStore = configureMockStore(middlewares);

  const initialState = { characters: { charactersList: [], isLoading: false } };

  beforeEach(() => {
    component = TestRenderer.create(
      <Provider store={mockStore(initialState)}>
        <House {...defaultProps}/>
      </Provider>
    );

    wrapper = component.root.findByType(View);
  });

  describe('when closed', () => {
    it('should render properly', () => {
      expect(wrapper.props.children).toMatchSnapshot();
    });
  });

  describe('when opened after clicking on it', () => {
    it('should render properly', () => {
      const header = component.root.findByType(Text);

      act(() => {
        header.props.onPress();
      });

      expect(wrapper.props.children).toMatchSnapshot();
    });
  });
});