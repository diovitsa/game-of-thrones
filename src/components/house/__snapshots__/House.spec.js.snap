// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`House when closed should render properly 1`] = `
Array [
  <Text
    id="header"
    onPress={[Function]}
    style={
      Object {
        "color": "blue",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    test house name
  </Text>,
  null,
]
`;

exports[`House when opened after clicking on it should render properly 1`] = `
Array [
  <Text
    id="header"
    onPress={[Function]}
    style={
      Object {
        "color": "blue",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    test house name
  </Text>,
  <View
    id="housesDetails"
  >
    <Text>
      Words: 
      testWords
    </Text>
    <Memo(Connect(CharacterList))
      house={
        Object {
          "name": "test house name",
          "words": "testWords",
        }
      }
    />
  </View>,
]
`;
